raiden-ts/src/transfers/reducer.ts

```diff
@@ -32,27 +32,25 @@ function transferSecretReducer(
   action: transferSecret | transferSecretRegister.success,
 ): RaidenState {
   const secrethash = action.meta.secrethash;
-  // store when seeing unconfirmed, but registerBlock only after confirmation
+
+  const transferState = state[action.meta.direction][secrethash];
+  if (!transferState) return state;
+
   const registerBlock =
     transferSecretRegister.success.is(action) && action.payload.confirmed
       ? action.payload.txBlock
-      : state[action.meta.direction][secrethash]?.secret?.[1]?.registerBlock ?? 0;
-  // don't overwrite registerBlock if secret already stored with it
-  if (
-    secrethash in state[action.meta.direction] &&
-    state[action.meta.direction][secrethash].secret?.[1]?.registerBlock !== registerBlock
-  )
-    state = {
-      ...state,
-      [action.meta.direction]: {
-        ...state[action.meta.direction],
-        [secrethash]: {
-          ...state[action.meta.direction][secrethash],
-          secret: timed({ value: action.payload.secret, registerBlock }),
-        },
+      : 0;
```

Comment:
This will reset the `registerBlock` if a `transferSecret` action (which doesn't contain a block) goes through, even if the block had been registered before. We don't expect many of these actions to come after a register, but it should be supported by the reducer and it should not lose this information.